hayeon/programmers-2/구명보트.js

```diff
@@ -0,0 +1,24 @@
+function solution(people, limit) {
+  let answer = 0;
+  people.sort((a, b) => a - b);
+
+  while (people.length > 0) {
+    const MaxWeightPeople = people.pop();
+    const MaxWeightBoat = limit - MaxWeightPeople;
+    if (people[0] <= MaxWeightBoat) {
+      people.splice(0, 1);
+    }
+    answer++;
+  }
+  return answer;
+}
```

Comment:
```suggestion
function solution(people, limit) {
  let answer = 0;
  people.sort((a, b) => a - b);

  while (people.length > 0) {
    const maxWeightPeople = people.pop();
    const maxWeightBoat = limit - maxWeightPeople;
    if (people[0] <= maxWeightBoat) {
      people.splice(0, 1);
    }
    answer++;
  }
  return answer;
}
```
- 풀이잘봤습니다 혼자 시도해보고 구글링 하는 것은 좋은데, 변수 네이밍 같은 부분은 다시 한번 검토를 통해서 수정해주세요 ~
- MaxWeightPeople, MaxWeightBoat 모두 파스칼 케이스로 작성되어 있습니다 카멜 케이스로 변경해주세요.
  - maxWeightPeople, maxWeightBoat
- 해당 풀이는 시간 복잡도로 접근했을 때 비효율적입니다. 아래와 같은 투 포인터를 이용하면 훨씬 효율적이게 접근할 수 있습니다.
```js
// 투포인터
function solution(people, limit) {
  let [start, end] = [0, people.length - 1];
  let result = 0;

  people.sort((a, b) => a - b);

  while (start <= end) {
    const sum = people[start] + people[end];

    if (sum <= limit) {
      start++;
      end--;
    } else if (sum > limit) {
      end--;
    }
    result++;
  }

  return result;
}
```
- 구글링하더라도 리팩토링 스스로 꼮 진행해주세요 🙏🙏🙏 x100